### DISCUSSION TITLE
Can YARP used to expose local services?

### DISCUSSION THREAD

**Discussion by savage8811** [dotnet/yarp#1378] [Q&A] [answered]
Is it possible to make a local service behind NAT accessible from internet? (like ngrok.com it does)

I’m thinking of a custom middleware which serialize the request-obj and send I through an existing socketconnection (SignalR?), the local instance will make the http-request and returns the response-obj through the socketconnection. Would that be conceivable?

**Comment by Tratcher** [comment, marked as answer]
A) ngrok.com looks like a pretty different service, designed for making connections from the inside out. It's possible, but I don't think you'd end up re-using much of YARP.
B) YARP is Level 7 (HTTP), not Level 4 (TCP), so only HTTP traffic would be relevant.
C) Setting up YARP behind a router is fine, but you'd have to do a more traditional port forwarding in the router to access it. From there, yes, it could route requests to internal HTTP services.

**Reply by savage8811** [reply]
thanks for the clarification
